### ISSUE TITLE
 Fix PUT response for eiger simulator

### ISSUE THREAD

**Issue by jsouter** [DiamondLightSource/tickit-devices#114]
I tested putting to every field returned by "keys" in each subsystem on a real Eiger 2 1M, the reponses are given in yaml below (for each parameter a list of updated parameters should be returend). Currently for most parameters the sim generally just returns a length 1 list with the name of parameter being modified. In a few cases the API is also broken, e.g. for `detector/api/1.8.0/config/threshold/1/mode`, only `['mode']` is returned instead of `['threshold/1/mode]`, at least on the eiger-stream2 branch. We should update the sim so that the PUT responses look more like what is given below. 

```
detector/api/1.8.0/config/auto_summation:
- auto_summation
- frame_count_time
detector/api/1.8.0/config/beam_center_x:
- beam_center_x
detector/api/1.8.0/config/beam_center_y:
- beam_center_y
detector/api/1.8.0/config/chi_increment:
- chi_increment
detector/api/1.8.0/config/chi_start:
- chi_start
detector/api/1.8.0/config/compression:
- compression
detector/api/1.8.0/config/count_time:
- bit_depth_image
- bit_depth_readout
- count_time
- countrate_correction_count_cutoff
- frame_count_time
- frame_time
detector/api/1.8.0/config/counting_mode:
- counting_mode
detector/api/1.8.0/config/countrate_correction_applied:
- countrate_correction_applied
detector/api/1.8.0/config/detector_distance:
- detector_distance
detector/api/1.8.0/config/detector_orientation:
- detector_orientation
detector/api/1.8.0/config/detector_translation:
- detector_translation
detector/api/1.8.0/config/extg_mode:
- extg_mode
detector/api/1.8.0/config/fast_arm:
- fast_arm
detector/api/1.8.0/config/flatfield:
- flatfield
- threshold/1/flatfield
detector/api/1.8.0/config/flatfield_correction_applied:
- flatfield_correction_applied
detector/api/1.8.0/config/frame_time:
- bit_depth_image
- bit_depth_readout
- count_time
- countrate_correction_count_cutoff
- frame_count_time
- frame_time
detector/api/1.8.0/config/incident_energy:
- element
- flatfield
- incident_energy
- photon_energy
- threshold/1/energy
- threshold/1/flatfield
- threshold/2/energy
- threshold/2/flatfield
- threshold_energy
- wavelength
detector/api/1.8.0/config/instrument_name:
- instrument_name
detector/api/1.8.0/config/kappa_increment:
- kappa_increment
detector/api/1.8.0/config/kappa_start:
- kappa_start
detector/api/1.8.0/config/mask_to_zero:
- mask_to_zero
detector/api/1.8.0/config/nexpi:
- nexpi
detector/api/1.8.0/config/nimages:
- nimages
detector/api/1.8.0/config/ntrigger:
- ntrigger
detector/api/1.8.0/config/ntriggers_skipped:
- ntriggers_skipped
detector/api/1.8.0/config/omega_increment:
- omega_increment
detector/api/1.8.0/config/omega_start:
- omega_start
detector/api/1.8.0/config/phi_increment:
- phi_increment
detector/api/1.8.0/config/phi_start:
- phi_start
detector/api/1.8.0/config/photon_energy:
- element
- flatfield
- incident_energy
- photon_energy
- threshold/1/energy
- threshold/1/flatfield
- threshold/2/energy
- threshold/2/flatfield
- threshold_energy
- wavelength
detector/api/1.8.0/config/pixel_mask:
- pixel_mask
- threshold/1/pixel_mask
detector/api/1.8.0/config/pixel_mask_applied:
- pixel_mask_applied
detector/api/1.8.0/config/roi_mode:
- count_time
- frame_time
- roi_mode
detector/api/1.8.0/config/sample_name:
- sample_name
detector/api/1.8.0/config/source_name:
- source_name
detector/api/1.8.0/config/test_image_mode:
- test_image_mode
detector/api/1.8.0/config/test_image_value:
- test_image_value
detector/api/1.8.0/config/threshold/1/energy:
- flatfield
- threshold/1/energy
- threshold/1/flatfield
- threshold/2/flatfield
- threshold_energy
detector/api/1.8.0/config/threshold/1/flatfield:
- flatfield
- threshold/1/flatfield
detector/api/1.8.0/config/threshold/1/mode:
- threshold/1/mode
- threshold/difference/mode
detector/api/1.8.0/config/threshold/1/pixel_mask:
- pixel_mask
- threshold/1/pixel_mask
detector/api/1.8.0/config/threshold/2/energy:
- flatfield
- threshold/1/flatfield
- threshold/2/energy
- threshold/2/flatfield
detector/api/1.8.0/config/threshold/2/flatfield:
- threshold/2/flatfield
detector/api/1.8.0/config/threshold/2/mode:
- threshold/2/mode
- threshold/difference/mode
detector/api/1.8.0/config/threshold/2/pixel_mask:
- threshold/2/pixel_mask
detector/api/1.8.0/config/threshold/difference/mode:
- difference_mode
detector/api/1.8.0/config/threshold_energy:
- flatfield
- threshold/1/energy
- threshold/1/flatfield
- threshold/2/flatfield
- threshold_energy
detector/api/1.8.0/config/trigger_mode:
- trigger_mode
detector/api/1.8.0/config/trigger_start_delay:
- trigger_start_delay
detector/api/1.8.0/config/two_theta_increment:
- two_theta_increment
detector/api/1.8.0/config/two_theta_start:
- two_theta_start
detector/api/1.8.0/config/virtual_pixel_correction_applied:
- virtual_pixel_correction_applied
filewriter/api/1.8.0/config/compression_enabled: []
filewriter/api/1.8.0/config/image_nr_start: []
filewriter/api/1.8.0/config/mode: []
filewriter/api/1.8.0/config/name_pattern: []
filewriter/api/1.8.0/config/nimages_per_file: []
monitor/api/1.8.0/config/buffer_size: []
monitor/api/1.8.0/config/discard_new: []
monitor/api/1.8.0/config/mode: []
stream/api/1.8.0/config/format: []
stream/api/1.8.0/config/header_appendix: []
stream/api/1.8.0/config/header_detail: []
stream/api/1.8.0/config/image_appendix: []
stream/api/1.8.0/config/mode: []
system/api/1.8.0/config/datetime/ntp: []
system/api/1.8.0/config/datetime/ntp_server: []
system/api/1.8.0/config/network/user1p1/dhcp: []
system/api/1.8.0/config/network/user1p1/mtu: []
system/api/1.8.0/config/network/user1p1/netmask: []
system/api/1.8.0/config/network/user1p1/static_routes: []
system/api/1.8.0/config/network/user1p2/addr: []

```

**Comment by GDYendell** [comment]
This is great, thanks @jsouter. I guess we need to add some more metadata to the fields to store a static list of parameters to return on put, rather than just itself.

**Comment by jsouter** [comment]
The strangest part is the response for `threshold/difference/mode` being `['difference_mode`]`, I wonder if that's the case in all API versions...

**Comment by GDYendell** [comment]
It's just because the logic in the http endpoint function pulls out the leaf of the URI to index down to the underlying threshold structure. It should reassemble it [here](https://github.com/DiamondLightSource/tickit-devices/pull/111/files#diff-f5c76a56a665ed999c30b4225acb24fa739379658030998d9117d106a30f4fbdR132) when it returns.

**Comment by GDYendell** [comment]
Fixed in #115
